### DISCUSSION TITLE
Creating relation between two tables with prisma

### DISCUSSION THREAD

**Discussion by JosephKorel** [prisma/prisma#15739] [Q&A]
i'm working in a NextJs project that uses prisma, but i never used it before. I created a relation between two fields, than i ran `yarn prisma generate` and retrieved it with `prisma.foo.findMany({})`. Everything worked fine at the local server, but i could not deploy it to vercel, i got an error saying that the field does not exist on that table.

I have the users table
```
model users {
  id                    Int       @id @default(autoincrement()) @db.UnsignedInt
  username              String?   @db.VarChar(255)
  email                 String    @db.VarChar(255)
  vehicle               Int?
  vehicleRef type_vehicles? @relation(fields: [vehicle], references: [id])
}
```

And i have the vehicle table

```
model type_vehicles {
  id         Int  @id @default(autoincrement()) @db.UnsignedInt
  name       String    @db.VarChar(255)

  users      users[ ]
}
```

The `vehicle` is the `id` of some vehicle of `type_vehicles`, so the field 'vehicle' references to the field 'id' of type_vehicles. Correct?

After that i ran `yarn prisma generate` and retrieved it as

```
const usersTmp = await prisma.users.findMany({
    select: {
      id: true,
      vehicleRef:{
            select:{
              name: true,
            },
          },
      updated_at: true,
 },
});

```

Am i doing something wrong? I'm only getting the error when deploying, saying that vehicleRef does not exist on users

Thanks in advance

**Comment by nurul3101** [comment]
Hello @JosephKorel 👋 

Based on your schema file, I tried to replicate it locally. You are trying to select `updated_at`, but it doesn't exist in the schema file.
Apart from it, this works locally.
```typescript
import { PrismaClient, Prisma } from '@prisma/client';

const prisma = new PrismaClient({
  log: ['query', 'info', 'warn'],
});

async function main() {
  const createUserAndVehicle = await prisma.users.create({
    data: {
      email: 'john@prisma.io',
      username: 'john',
      vehicleRef: {
        create: {
          name: 'car',
        },
      },
    },
  });

  const usersTmp = await prisma.users.findMany({
    select: {
      id: true,
      vehicleRef: {
        select: {
          name: true,
        },
      },
    },
  });

  console.log(usersTmp);
}

main()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });
```
Response:
```
> ts-node index.ts

prisma:info Starting a postgresql pool with 0 connections.
prisma:query BEGIN
prisma:query INSERT INTO "public"."type_vehicles" ("name") VALUES ($1) RETURNING "public"."type_vehicles"."id" /* traceparent=00-00-00-00 */
prisma:query INSERT INTO "public"."users" ("username","email","vehicle") VALUES ($1,$2,$3) RETURNING "public"."users"."id" /* traceparent=00-00-00-00 */
prisma:query SELECT "public"."users"."id", "public"."users"."username", "public"."users"."email", "public"."users"."vehicle" FROM "public"."users" WHERE "public"."users"."id" = $1 LIMIT $2 OFFSET $3 /* traceparent=00-00-00-00 */
prisma:query COMMIT
prisma:query SELECT "public"."users"."id", "public"."users"."vehicle" FROM "public"."users" WHERE 1=1 OFFSET $1 /* traceparent=00-00-00-00 */
prisma:query SELECT "public"."type_vehicles"."id", "public"."type_vehicles"."name" FROM "public"."type_vehicles" WHERE "public"."type_vehicles"."id" IN ($1) OFFSET $2
[ { id: 1, vehicleRef: { name: 'car' } } ]
```

Can you check if the `prisma generate` command is included in the build step of your package.json?
Here's a reference of package.json file of [vercel deployment example](https://github.com/prisma/deployment-example-vercel/blob/main/package.json)

**Reply by JosephKorel** [reply]
Yes, it's included in the build step. So to make a relation i have to use `prisma.users.create` ? I can't just make the relation at `schema.prisma` file?

**Reply by nurul3101** [reply]
No you don't need to use `prisma.users.create` to create a relation. You can define it in schema file.

**Reply by pantharshit00** [reply]
👋 Hello @JosephKorel,  did you perhaps have a chance to check @nurul3101's recommendation? Let us know if you still have any questions!
 
 Or, if you've found a solution, we'd love to know how you solved this problem. Feel free to post it in this thread and "Mark as answer" if this discussion is now solved!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
